UPGRADING.md

```diff
@@ -243,6 +243,31 @@ state = State(networks={"foo": Network.default()})
 state = State(networks={Network.default("foo")})
 ```
 
+### Use the .deferred() method to populate State.deferred
+
+Previously, there were multiple methods to populate the `State.deferred` list:
+events with a `.deferred()` method, the `scenario.deferred()` method, and
+creating a `DeferredEvent` object manually. Now, for Juju events, you should
+always use the `.deferred()` method of the event - this also ensures that the
+deferred event has all of the required links (to relations, containers, secrets,
+and so on).
+
```

Comment:
nitpick: The dash before "this also ensures that..." should technically be an [em-dash](https://www.merriam-webster.com/grammar/em-dash-en-dash-how-to-use) or at least an approximation thereof in the form of a double dash with spaces around. In Juju docs I do the latter. So: ".. of the event -- this also ..."